### PR TITLE
EPMDJ-3322: No success message when enabling / disabling agent

### PR DESCRIPTION
**Pull request by SMPlay**
Issue: EPMDJ-3322

### CODE REVIEW

File: src/pages/agents-page/agents-table/agents-table.tsx [hunk 1]
```diff
@@ -18,71 +19,92 @@ interface Props {
...
-        <Column name="expander" Cell={() => null} />,
-        <Column
-          name="name"
-          label="Name"
-          Cell={({ item }) => <NameColumn agent={item} withMargin />}
-        />,
+export const AgentsTable = agentsTable(({ className, agents }: Props) => {
+  const { showMessage } = React.useContext(NotificationManagerContext);
+
+  const onChangeAgentStatus = async (agentId: string, status: string) => {
```
Comment (RomanAchilov) [inline src/pages/agents-page/agents-table/agents-table.tsx:25]: async/await need here?

File: src/pages/agents-page/agents-table/agents-table.tsx [hunk 2]
```diff
@@ -18,71 +19,92 @@ interface Props {
...
+    await toggleAgent(
+      agentId,
+      () => {
+        showMessage({
+          type: 'SUCCESS',
+          text: `Agent has been ${status === AGENT_STATUS.ONLINE ? 'disabled' : 'enabled'}.`,
+        });
+      },
+      (errorMessage: string) => {
```
Comment (RomanAchilov) [inline src/pages/agents-page/agents-table/agents-table.tsx:34]: ` (errorMessage: string) => showMessage({
          type: 'ERROR',
          text: errorMessage || 'There is some issue with your action. Please try again later.',
        })`

File: src/pages/agents-page/agents-table/agents-table.tsx [hunk 3]
```diff
@@ -18,71 +19,92 @@ interface Props {
...
+            label="Environment"
+            Cell={({ value, item }) => (
+              <span>{item.status === AGENT_STATUS.NOT_REGISTERED ? 'n/a' : value}</span>
+            )}
+          />,
+          <Column
+            name="status"
+            label="Status"
+            Cell={({ value, item }) => (
+              <AgentStatusToggler status={value} onChange={() => toggleAgent(item.id)} />
```
Comment (RomanAchilov) [inline src/pages/agents-page/agents-table/agents-table.tsx:72]: toggle for service group agents should not work?

**Review by RomanAchilov — CHANGES_REQUESTED** [review]
1) check https://jiraeu.epam.com/browse/EPMDJ-3322
2) rebase

**Review by RomanAchilov — APPROVED** [review]

File: src/pages/settings-page/settings-page.tsx [hunk 4]
```diff
@@ -30,6 +30,21 @@ export const SettingsPage = settingsPage(
     const data = useCommonEntity(id, type) || {};
     const { showMessage } = React.useContext(NotificationManagerContext);
     const [isUnregisterModalOpen, setIsUnregisterModalOpen] = React.useState(false);
+
+    const onChangeAgentStatus = (agentId: string, status: string) => {
```
Comment (pavyarov) [inline src/pages/settings-page/settings-page.tsx:34]: completely copy-paste

**Review by pavyarov — CHANGES_REQUESTED** [review]

File: src/pages/settings-page/settings-page.tsx (hunk 4 above)
Comment (SMPlay) [reply src/pages/settings-page/settings-page.tsx:34]: Should I just remove the handler or add success and error logic to the toggle-agent function?

**Review by RomanAchilov — APPROVED** [review]

File: src/pages/api/toggle-agent.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import axios from 'axios';
+
+import { Message } from 'types/message';
+import { AGENT_STATUS } from 'common/constants';
+
+export async function toggleAgent(
+  agentId: string,
+  status: string,
```
Comment (pavyarov) [inline src/pages/api/toggle-agent.ts:8]: agent `status` is definitely not string

**Review by pavyarov — CHANGES_REQUESTED** [review]

File: src/pages/api/toggle-agent.ts (hunk 5 above)
Comment (SMPlay) [reply src/pages/api/toggle-agent.ts:8]: `status` gets a string `ONLINE` or `OFFLINE`

**Review by pavyarov — APPROVED** [review]

**Pull request merged**